test/org/sagebionetworks/bridge/services/ConsentServiceMockTest.java

```diff
@@ -407,6 +427,23 @@ public void withdrawFromStudyWithPhone() {
         verify(notificationsService).deleteAllRegistrations(study.getStudyIdentifier(), HEALTH_CODE);
     }
     
+    @Test
+    public void withdrawFromStudyRemovesDataGroups() {
+        account.setConsentSignatureHistory(SUBPOP_GUID, ImmutableList.of(CONSENT_SIGNATURE));
+
+        account.setDataGroups(new HashSet<>(TestConstants.USER_DATA_GROUPS));
```

Comment:
Add additional data groups, so only the ones specified by the subpopulation are removed.

Reply:
Added